WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -48,6 +51,9 @@ def create
         @competition.errors[:competition_id_to_clone] = "invalid"
       end
     end
+    if current_user.any_kind_of_delegate?
+      @competition.delegates << current_user
+    end
 
     if @competition.errors.size == 0 && @competition.save
```

Comment:
Got it! I was able to push this down into the Competition model in
https://github.com/jfly/worldcubeassociation.org/commit/680a4f1f57636e17788654cd546b3e6c6726bb35
.

On Wed, Sep 23, 2015 at 2:27 AM, Tim Habermaas <notifications@github.com>
wrote:

> In WcaOnRails/app/controllers/competitions_controller.rb
> <https://github.com/cubing/worldcubeassociation.org/pull/176#discussion_r40184555>
> :
>
> > @@ -48,6 +51,9 @@ def create
> >          @competition.errors[:competition_id_to_clone] = "invalid"
> >        end
> >      end
> > +    if current_user.any_kind_of_delegate?
> > +      @competition.delegates << current_user
> > +    end
> >
> >      if @competition.errors.size == 0 && @competition.save
>
> Oh, totally missed this. [image: :sob:]
>
> Moving it to the competition model sounds reasonable. I'm not sure how
> right now, though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cubing/worldcubeassociation.org/pull/176/files#r40184555>
> .
>